tracee/tracee.go

```diff
@@ -812,6 +814,49 @@ func (t *Tracee) shouldProcessEvent(e RawEvent) bool {
 
 func (t *Tracee) processEvent(ctx *context, args map[argTag]interface{}) error {
 	switch ctx.EventID {
+
+	//capture written files
+	case VfsWriteEventID, VfsWritevEventID:
+		if t.config.CaptureWrite {
+			filePath, ok := args[t.EncParamName[ctx.EventID%2]["pathname"]].(string)
+			if !ok {
+				return fmt.Errorf("error parsing vfs_write args")
+			}
+			// path should be absolute, except for e.g memfd_create files
+			if filePath == "" || filePath[0] != '/' {
+				return nil
+			}
+			dev, ok := args[t.EncParamName[ctx.EventID%2]["dev"]].(int32)
+			if !ok {
+				return fmt.Errorf("error parsing vfs_write args")
+			}
+			inode, ok := args[t.EncParamName[ctx.EventID%2]["inode"]].(int32)
+			if !ok {
+				return fmt.Errorf("error parsing vfs_write args")
+			}
+
+			// stop processing if already write was already indexed
+			fileName := fmt.Sprintf("write.dev-%d.inode-%d", dev, inode)
+			indexName, ok := t.writtenFiles[fileName]
+			if ok && indexName == filePath {
+				return nil
+			}
+
+			// write original filepath mapping to file
+			destinationFilePath := filepath.Join(t.config.OutputPath, "written_files")
+			f, err := os.OpenFile(destinationFilePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				return fmt.Errorf("error logging written files")
+			}
+			defer f.Close()
```

Comment:
open/close for every processed event can introduce unnecessary overhead. I think it would be better to just collect the index here, and actually print it at then end of the run. wdyt?

Reply:
Agree, that will be more efficient

Reply:
Assuming this should likely be implemented in `eventPrinter.Epilogue()` in `printer.go` with the stats of the run?

Reply:
actually I wouldn't put it in a `eventPrinter` because it is not output format dependent (it's the same code regardless of the chosen output format, just like the other artifacts). You can call it from `Run` https://github.com/aquasecurity/tracee/blob/main/tracee/tracee.go#L725